tests/integration/examples/example_validations.py

```diff
@@ -1,107 +0,0 @@
-from zenml.post_execution import get_pipeline
-
-
-def caching_example_validation():
-    """Validates the stored pipeline run info after running the caching example."""
-    pipeline = get_pipeline("mnist_pipeline")
-    assert pipeline
-
-    first_run, second_run = pipeline.runs[-2:]
-
-    # Both runs should be completed
-    assert first_run.status == ExecutionStatus.COMPLETED
-    assert second_run.status == ExecutionStatus.COMPLETED
-
-    # The first run should not have any cached steps
-    for step in first_run.steps:
-        assert not step.is_cached
-
-    # The second run should have two cached steps (chronologically first 2)
-    assert second_run.steps[0].is_cached
-    assert second_run.steps[1].is_cached
-    assert not second_run.steps[2].is_cached
-    assert not second_run.steps[3].is_cached
-
-
-def mlflow_deployment_example_validation():
-    """Validates the stored pipeline run info after running the MLflow deployment example."""
-    # Verify the example run was successful
-    deployment_pipeline = get_pipeline("continuous_deployment_pipeline")
-    assert deployment_pipeline is not None
-
-    inference_pipeline = get_pipeline("inference_pipeline")
-    assert inference_pipeline is not None
-
-    deployment_run = deployment_pipeline.runs[-1]
-
-    # Run should be completed
-    assert deployment_run.status == ExecutionStatus.COMPLETED
-
-    for step in deployment_run.steps:
-        assert step.status == ExecutionStatus.COMPLETED
-
-    inference_run = inference_pipeline.runs[-1]
-
-    # Run should be completed
-    assert inference_run.status == ExecutionStatus.COMPLETED
-
-    for step in inference_run.steps:
-        assert step.status == ExecutionStatus.COMPLETED
-
-    import mlflow
-    from mlflow.tracking import MlflowClient
-
-    from zenml.integrations.mlflow.mlflow_environment import MLFlowEnvironment
-
-    # create and activate the global MLflow environment
-    MLFlowEnvironment().activate()
-
-    # fetch the MLflow experiment created for the pipeline runs
-    mlflow_experiment = mlflow.get_experiment_by_name(deployment_pipeline.name)
-
-    assert mlflow_experiment is not None
-
-    # fetch all MLflow runs created for the pipeline
-    mlflow_runs = mlflow.search_runs(
-        experiment_ids=[mlflow_experiment.experiment_id], output_format="list"
-    )
-    assert len(mlflow_runs) == 1
-
-    # fetch the MLflow run created for the pipeline run
-    mlflow_runs = mlflow.search_runs(
-        experiment_ids=[mlflow_experiment.experiment_id],
-        filter_string=f'tags.mlflow.runName = "{deployment_run.name}"',
-        output_format="list",
-    )
-    assert len(mlflow_runs) == 1
-    mlflow_run = mlflow_runs[0]
-
-    client = MlflowClient()
-    # fetch the MLflow artifacts logged during the first pipeline run
-    artifacts = client.list_artifacts(mlflow_run.info.run_id)
-    assert len(artifacts) == 3
-
-    from zenml.integrations.mlflow.services import MLFlowDeploymentService
```

Comment:
I kept this around for so long in the hope that we could reuse some of this to test the mlflow deployer example. Is it really un-salvageable ?

Reply:
No, it was just my lazy brain that decided to ignore it :P
I added the parts that made sense to the new validation